### DISCUSSION TITLE
agent: can't detect the kernels

### DISCUSSION THREAD

**Discussion by youngjun0627** [lablup/contribution-academy-2021#100] [Show and tell]
- commit
[agent-309 commit](https://github.com/lablup/backend.ai-agent/pull/309)

- problem
해당 커밋에서는 kernel-host를 bind-host로 수정하고 다른 파일의 키값을 수정하여 보완하였습니다. 그러나 해당 커밋이 merge된 이후 커널을 제대로 잡지 못하는 문제가 발생하여 커널이 실행되고 추적하지 못하는 현상이 발생한것으로 확인되었습니다(저의 생각입니다). 따라서 저는 해당 문제를 해결하기 위해 파일들을 수정하고 [PR](https://github.com/lablup/backend.ai-agent/pull/310)을 작성한 상태입니다. 자세한 내용은 해당 PR을 참고하시길 바랍니다. 또한 해당 커밋이 현재 진행중인지 확인이 필요합니다.

- solution
현재 agent에 최근에 merge된 커밋에서 toml 파일들을 수정했는데요. 현재 agent폴더에 있는 agent.toml파일과 맞지 않아 agent가 커널을 제대로 잡지 못하는 에러가 발생했습니다(이미 install된 경우에 해당합니다). 이와 관련해서 PR요청했는데 혹시 확인 가능하신분은 [PR](https://github.com/lablup/backend.ai-agent/pull/310)에 있는 파일처럼 코드 변경하시고, agent/config/halfstack.toml 파일을 agent/agent.toml(etcd, agent, watcher 포트번호 이전의 agent.toml파일의 내용으로 변경해야됌)로 copy하신 후 확인해주세요.

**Comment by mindgitrwx** [comment]
/backend.ai-dev/agent/src/ai/backend/agent/config.py 파일의 36번째 라인에 
`t.key('kernel-host', default=''): t.String(allow_blank=True)`
를 넣어도 실행이 되는 것으로 보입니다.

`git reset --hard $(git rev-parse @~)` 으로 바로 전 커밋내역으로 돌려서 실행해도 됩니다.

**Reply by youngjun0627** [reply]
최근 커밋으로 문제가 발생한건 확실한거 같네용

**Comment by lizable** [comment]
I'll investigate it 'til this week. It's also happening to me as well...

**Reply by youngjun0627** [reply]
I make a [PR](https://github.com/lablup/backend.ai-agent/pull/310). After reading this, give me feedback!

**Comment by youngjun0627** [comment]
This issue is resolved.
ref: [#310](https://github.com/lablup/backend.ai-agent/pull/310)
